### PR TITLE
Drops react-hmre in favor of custom setup allowing for createClass alone

### PR DESCRIPTION
**Pull request by rcsole**
Enables hot reloading for `createClass` and `React.createClass`.

I'm working on getting rid of the warning thrown by `Router`. So this
a work in progress, but I'd like feedback on this part @dmbch.

### CODE REVIEW

**Comment by rcsole** [comment]
Also this: https://github.com/reactjs/react-router/pull/3371/files

File: package.json [hunk 1]
```diff
@@ -79,6 +79,9 @@
     "react-redux": "4.4.5",
     "react-router": "2.3.0",
     "react-router-redux": "4.0.4",
+    "react-transform-catch-errors": "1.0.2",
+    "react-transform-hmr": "1.0.4",
+    "redbox-react": "1.2.3",
```
Comment (dmbch) [inline package.json]: I'm not sure I like this one: I'm just not so keen on in-band debug output and personally prefer it out-of-band, i.e. in the console.

Comment (rcsole) [inline package.json]: 👍

**Comment by rcsole** [comment]
As discussed I think I'll close this for now. Let's look at how [RHL3](https://github.com/gaearon/react-hot-loader/tree/next) evolves.

**Comment by rudeworks** [comment]
RHL3 is now in beta https://twitter.com/dan_abramov/status/726458501556502528

**Comment by dmbch** [comment]
Nice :)

**Pull request closed without merging**